# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/svg/nsFilterInstance.h
# Commit: 3c9634352cd6
# Full Hash: 3c9634352cd66e1debc3beeedd53130d16474db2
# Author: Jeff Muizelaar <jrmuizel@gmail.com>
# Date: 2019-01-28 04:51:03
# Regressor Bug: 1522021
# File Overlap Count: 1
# Description:
#   Bug 1522021. Propagate input size to CreateSimilarDrawTarget for filters. r=mstange
#   
#   This lets us avoid drawing the complete input for ever tile when
#   drawing filters into a tile.
#   
# ==============================================================================

diff -r 747303cab27a -r 3c9634352cd6 layout/svg/nsFilterInstance.h
--- a/layout/svg/nsFilterInstance.h	Sun Jan 27 22:06:38 2019 +0000
+++ b/layout/svg/nsFilterInstance.h	Sun Jan 27 23:36:04 2019 +0000
@@ -240,7 +240,10 @@
    * Creates the SourceSurface for the SourceGraphic graph node, paints its
    * contents, and assigns it to mSourceGraphic.mSourceSurface.
    */
-  void BuildSourceImage(DrawTarget* aDest, imgDrawingParams& aImgParams);
+  void BuildSourceImage(DrawTarget* aDest, imgDrawingParams& aImgParams,
+                        mozilla::gfx::FilterNode* aFilter,
+                        mozilla::gfx::FilterNode* aSource,
+                        const mozilla::gfx::Rect& aSourceRect);
 
   /**
    * Build the list of FilterPrimitiveDescriptions that describes the filter's
